### PR TITLE
Update snippet creation bool to env var only

### PR DESCRIPTION
**Pull request by jobselko**
Obsoleted draft

### CODE REVIEW

File: collectors/nvd/collectors.py [hunk 1]
```diff
@@ -250,7 +244,7 @@ def collect(self, cve: Union[str, None] = None) -> str:
 
             # create a new snippet with a flaw (if it does not exist) and link them
             snippet = None
-            if self.snippet_creation_enabled:
+            if SNIPPET_CREATION_ENABLED:
```
Comment (jobselko) [inline collectors/nvd/collectors.py:247]: @Elkasitu I am struggling with removing the `snippet_creation_enabled` instance attribute and replacing it with `SNIPPET_CREATION_ENABLED` env var only. The code works when testing it with `SNIPPET_CREATION=1`/`SNIPPET_CREATION=0` from the `.env` file, but it fails in `collectors/nvd/tests/test_collectors.py`.

It seems that the `snippet_creation_enabled` fixture is correctly set in `test_snippet_and_flaw_created()` in `collectors/nvd/tests/test_collectors.py`, but in `collectors/nvd/collectors.py`, this value is lost after the import from `collectors/constants.py`.

Do you have any idea how to solve this?

Comment (jobselko) [reply collectors/nvd/collectors.py:247]: So, while this change does not work in my local dev env, it works in CI :D

File: collectors/constants.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 from osidb.helpers import get_env
 
+# todo?
```
Comment (osoukup) [inline collectors/constants.py:3]: What is the meaning of this?

File: collectors/nvd/collectors.py (hunk 1 above)
Comment (osoukup) [reply collectors/nvd/collectors.py:247]: I do not understand the changes in this file. Functionally they do nothing new.

**Review by osoukup — CHANGES_REQUESTED** [review]
This seems to be a draft but it is not marked so.

File: collectors/nvd/collectors.py (hunk 1 above)
Comment (jobselko) [reply collectors/nvd/collectors.py:247]: @osoukup It is a follow-up to Adrian's https://github.com/RedHatProductSecurity/osidb/pull/379#issuecomment-1810211075.

So far, I updated only one test, but its result differs depending on the environment (passing in CI / failing locally). This is not ready for merging but I would like to get some feedback regarding what might be a problem here.

File: collectors/constants.py (hunk 2 above)
Comment (jobselko) [reply collectors/constants.py:3]: I tried some `getenv` things here, but you can ignore it. It is mostly for visibility now.

File: collectors/nvd/collectors.py (hunk 1 above)
Comment (Elkasitu) [reply collectors/nvd/collectors.py:247]: You have to monkeypatch the constant and not the env var, the constant's value remains static for the lifecycle of the test run regardless of changes to the env var

Comment (jobselko) [reply collectors/nvd/collectors.py:247]: The result is still same - not working locally. I tried to use both `1` and `True`.

Comment (jobselko) [reply collectors/nvd/collectors.py:247]: @Elkasitu I would prefer to leave the attr as it is now as this change seems to be not so important and hard to implement (or at least for me). WDYT?

Comment (Elkasitu) [reply collectors/nvd/collectors.py:247]: Sure

**Pull request closed without merging**